src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -213,6 +216,60 @@ internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<byte> byte
             }
         }
 
+        static readonly byte[] _singleZeroByteArray = new byte[1] { 0 };
+
+        internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<ConstantValue> constants, SourceHashAlgorithm hashAlgorithm = SourceHashAlgorithms.Default)
+        {
+            var algorithmName = GetAlgorithmName(hashAlgorithm);
+            using var incrementalHash = IncrementalHash.CreateHash(algorithmName);
+
+            foreach (var constant in constants)
+            {
+                incrementalHash.AppendData(getBytes(constant));
+            }
+
+            return ImmutableArray.Create(incrementalHash.GetHashAndReset());
+
+            static byte[] getBytes(ConstantValue constant)
+            {
+                Debug.Assert(Enum.GetValues(typeof(ConstantValueTypeDiscriminator)).Cast<ConstantValueTypeDiscriminator>().Max()
+                    == ConstantValueTypeDiscriminator.DateTime);
+
+                switch (constant.Discriminator)
+                {
+                    case ConstantValueTypeDiscriminator.Nothing:
+                        return _singleZeroByteArray;
+
+                    case ConstantValueTypeDiscriminator.String:
+                        return Encoding.Unicode.GetBytes(constant.StringValue!);
+
+                    case ConstantValueTypeDiscriminator.NInt:
+                        return BitConverter.GetBytes(constant.Int32Value);
```

Comment:
>BitConverter.GetBytes(constant.Int32Value) [](http://example.com/codeflow?start=31&length=42)

The order of bytes in the array returned by the GetBytes method depends on whether the computer architecture is little-endian or big-endian. So, build on different machines might produce different bytes for the same value. Is this a concern given the purpose of this helper?